x-pack/metricbeat/module/openai/usage/persistcache.go

```diff
@@ -0,0 +1,172 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package usage
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"os"
+	"path"
+	"strings"
+	"sync"
+	"time"
+)
+
+// stateManager handles the storage and retrieval of state data with key hashing and caching capabilities
+type stateManager struct {
+	store     *stateStore
+	keyPrefix string
+	hashCache sync.Map
+}
+
+// stateStore handles persistence of key-value pairs using the filesystem
+type stateStore struct {
+	sync.RWMutex        // Protects access to the state store
+	Dir          string // Base directory for storing state files
+}
+
+// newStateManager creates a new state manager instance with the given storage path
+func newStateManager(storePath string) (*stateManager, error) {
+	if strings.TrimSpace(storePath) == "" {
+		return nil, errors.New("empty path provided")
+	}
+
+	store, err := newStateStore(storePath)
```

Comment:
Do we need to check if the storepath indeed exists ?

Reply:
we are creating the storePath. Part of it already there and the rest we create with MkdirAll.

Reply:
See this: `newStateManager(paths.Resolve(paths.Data, path.Join("state", base.Module().Name(), base.Name())))`

paths.Data is supplied by Agent using `--paths.data` flag. So, even if the directory doesn't exist, we create them.